### PR TITLE
fix(rubocop): make the cop happy

### PR DESCRIPTION
**Pull request by baby-gnu**
Fix every problem reported by rubycop.

### CODE REVIEW

**Comment by baby-gnu** [comment]
When the rubycop will be happy I'll add a new commit to move the job with others.

**Comment by baby-gnu** [comment]
Hello @myii 

I did not do the `--safe-auto-correct` because I wanted to understand the errors to avoid making them again.

I was mostly annoyed by the `method_missing` hack and try several things before discovering inspec/inspec#4328.

I prefer having one commit per fix, the branch is here to summary the work.

Regards.

**Comment by baby-gnu** [comment]
Maybe should I rewrote all the subject since it's `rubocop` and not `rubycop` :-/

I'll do that with the last commit to merge lint and rubocop job.

Regards.

**Comment by myii** [comment]
> I prefer having one commit per fix, the branch is here to summary the work.

No problem at all.

> Maybe should I rewrote all the subject since it's `rubocop` and not `rubycop` :-/

It's strange, I find `rubocop` really difficult to spell -- my fingers keep trying to type different things!

**Comment by baby-gnu** [comment]
Much better when using the same `rubocop` version locally than on travis.

File: test/integration/default/controls/config_spec.rb [hunk 1]
```diff
@@ -16,31 +15,30 @@
...
+  'auth_tcp' => 'none'
 }
 
 control 'Libvirt configuration' do
   title 'verify applied configuration'
 
   describe libvirt.daemon_config_file do
     it { should exist }
-    its('content') { should match %r{This\sfile\sis\smanaged\sby\sSalt} }
+    its('content') { should match /This\sfile\sis\smanaged\sby\sSalt/ }
```
Comment (myii) [inline test/integration/default/controls/config_spec.rb]:
```suggestion
    its('content') { should match(/This\sfile\sis\smanaged\sby\sSalt/) }
```

File: test/integration/default/controls/config_spec.rb [hunk 2]
```diff
@@ -16,31 +15,30 @@
...
   end
 
   describe file('/etc/libvirt/libvirtd.conf') do
     it { should exist }
-    its('content') { should match %r{This\sfile\sis\smanaged\sby\sSalt} }
+    its('content') { should match /This\sfile\sis\smanaged\sby\sSalt/ }
```
Comment (myii) [inline test/integration/default/controls/config_spec.rb]:
```suggestion
    its('content') { should match(/This\sfile\sis\smanaged\sby\sSalt/) }
```

File: .rubocop.yml [hunk 3]
```diff
@@ -7,4 +7,10 @@ Metrics/LineLength:
   # Based on https://github.com/PyCQA/flake8-bugbear#opinionated-warnings (`B950`)
   Max: 88
 
+# libvirt-formula overredies
+Lint/AmbiguousRegexpLiteral:
+  # Exclude { should match /<REGEX>/ }
+  Exclude:
+    - 'test/integration/default/controls/config_spec.rb'
+
```
Comment (myii) [inline .rubocop.yml]: This whole exclusion section won't  be required with the two fixes I've proposed.

**Comment by myii** [comment]
Thanks for the work @baby-gnu, I'll add the changes in a separate PR.

**Pull request merged**